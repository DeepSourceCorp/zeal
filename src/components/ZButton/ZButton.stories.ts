import ZButton from './ZButton.vue'

import { action } from '@storybook/addon-actions'

import '../../assets/css/tailwind.css'
import '../../assets/css/typography.css'
import '../../assets/css/layout.css'

export default {
  title: 'Zeal/Button',
  component: ZButton,
  excludeStories: /.*Data$/
}

export const DefaultWithoutText = () => ({
  components: { ZButton },
  template: '<z-button></z-button>'
})

export const ButtonWithText = () => ({
  components: { ZButton },
  template: '<z-button color="primary">Hello World</z-button>'
})

export const Colors = () => ({
  components: { ZButton },
  template: `<div>
        <z-button color="primary">Hello World</z-button>
        <z-button color="secondary">Hello World</z-button>
    </div>`
})

export const Sizes = () => ({
  components: { ZButton },
  template: `<div>
        <z-button size="small">Small Button</z-button>
        <z-button size="medium">Medium Button</z-button>
        <z-button size="large">Large Button</z-button>
        <z-button size="xlarge">Extra Large Button</z-button>
    </div>`
})

export const States = () => ({
  components: { ZButton },
  template: `<div>
        <z-button active>Active Button</z-button>
        <z-button disabled>Disabled Button</z-button>
        <z-button color="secondary" active>Active Secondary Button</z-button>
        <z-button color="secondary" disabled>Disabled Secondary Button</z-button>
    </div>`
})

export const FullWidthButton = () => ({
  components: { ZButton },
  template: `<div>
        <z-button full-width>Full Width Button</z-button>
    </div>`
})

export const DefaultLinkButton = () => ({
  components: { ZButton },
  template: `<div>
        <z-button color="link" type="link" to="https://deepsource.io/">Link Button</z-button>
    </div>`
})

export const ActiveLinkButton = () => ({
  components: { ZButton },
  template: `<div>
        <z-button color="link" type="link" to="https://deepsource.io/" active>Link Button</z-button>
    </div>`
})

export const DisabledLinkButton = () => ({
  components: { ZButton },
  template: `<div>
        <z-button color="link" type="link" to="https://deepsource.io/" disabled>Link Button</z-button>
    </div>`
})
