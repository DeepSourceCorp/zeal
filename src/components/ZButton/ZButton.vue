<template>
  <component
    :is="type === 'link' || to ? 'a' : 'button'"
    :href="to"
    class="z-btn"
    v-bind="$attrs"
    :disabled="disabled"
    v-on="$listeners"
    :class="[
      `${color && `z-btn--${color}`}`,
      `${size && `z-btn--${size}`}`,
      {
        'is-disabled': this.disabled !== false,
        'is-active': this.active !== false,
        'full-width': this.fullWidth !== false
      },
      customClasses,
      `${!color && !customClasses ? 'z-btn--primary' : ''}`
    ]"
    :type="type"
  >
    <slot>Click</slot>
  </component>
</template>

<script>
export default {
  name: 'ZButton',
  props: {
    color: {
      default: '',
      type: String
    },
    type: {
      default: 'button',
      type: String
    },
    customClasses: {
      default: '',
      type: String
    },
    size: {
      default: '',
      type: String
    },
    disabled: {
      default: false,
      type: Boolean
    },
    active: {
      default: false,
      type: Boolean
    },
    fullWidth: {
      default: false,
      type: Boolean
    },
    to: {
      default: '',
      type: String
    }
  },
  methods: {
    handleClick(event) {
      this.$emit('click', event)
    }
  }
}
</script>

<style lang="css" scoped>
/* Common Button Style */

.z-btn {
  @apply inline-flex items-center px-6 text-base font-medium border-transparent h-10 rounded-sm relative;
}

.is-disabled {
  @apply opacity-50 cursor-not-allowed;
}

.full-width {
  @apply w-full inline-block;
}

/* Button - Primary Style */

.z-btn--primary {
  @apply bg-juniper hover:bg-light_juniper text-ink-400;
}

.z-btn--primary.is-disabled {
  @apply hover:bg-juniper;
}

/* Button - Secondary Style */

.z-btn--secondary {
  @apply bg-ink-300 hover:bg-ink-200 text-vanilla-100 border border-slate;
}

.z-btn--secondary.is-disabled {
  @apply hover:bg-ink-300;
}

.z-btn--secondary.is-active {
  @apply bg-ink-100;
}

.z-btn--link {
  @apply font-normal text-juniper hover:underline;
}

.z-btn--link.is-active {
  @apply underline;
}

.z-btn--link.is-disabled {
  @apply text-slate no-underline;
}

/* Different Sizes of the Button*/
.z-btn--small {
  @apply h-8 text-xs;
}

.z-btn--medium {
  @apply h-10 text-base;
}

.z-btn--large {
  @apply h-12 text-lg;
}

.z-btn--xlarge {
  @apply h-16 text-lg;
}
</style>
