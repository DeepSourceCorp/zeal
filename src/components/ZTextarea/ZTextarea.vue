<template>
  <div
    class="h-full overflow-scroll p-1 border border-ink-200 flex items-start focus-within:shadow-white rounded-sm"
    :class="{
      'cursor-not-allowed': disabled,
      'resize-none': !resizable,
      resize: resizable
    }"
  >
    <textarea
      class="w-full h-full p-1 bg-transparent resize-none outline-none"
      :class="[
        `text-${textSize}`,
        disabled ? 'text-slate' : `text-${textColor}`,
        {
          'cursor-not-allowed': disabled
        }
      ]"
      :placeholder="placeholder"
      :disabled="disabled"
      :value="value"
      @input="updateSelf($event.target.value)"
    >
    </textarea>
    <slot></slot>
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
export default Vue.extend({
  name: 'ZTextarea',
  props: {
    value: {
      default: '',
      type: String
    },
    placeholder: {
      default: 'Once upon a time...',
      type: String
    },
    textSize: {
      type: String,
      default: 'xs'
    },
    textColor: {
      type: String,
      default: 'vanilla-400'
    },
    disabled: {
      type: Boolean,
      default: false
    },
    resizable: {
      type: Boolean,
      default: false
    }
  },
  model: {
    prop: 'value',
    event: 'input'
  },

  methods: {
    updateSelf(value: string): void {
      if (!this.disabled) this.$emit('input', value)
    }
  }
})
</script>
