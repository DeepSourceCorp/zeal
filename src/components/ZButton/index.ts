import ZButton from './ZButton.vue';

export { ZButton };
