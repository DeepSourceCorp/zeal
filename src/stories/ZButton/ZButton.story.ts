import ZButton from '../../components/ZButton/ZButton.vue'
import { Meta, Story } from '@storybook/addon-docs/blocks'
import { component } from 'vue/types/umd'

export const ButtonWithText = () => ({
  components: { ZButton },
  template: '<z-button>Hello Button</button>',
  title: 'ZButton'
})

ButtonWithText.story = {
  name: 'Button with Text'
}
