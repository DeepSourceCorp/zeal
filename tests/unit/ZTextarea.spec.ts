import ZTextarea from '../../src/components/ZTextarea'
import ZButton from '../../src/components/ZButton'
import Vue from 'vue'

import { mount, Wrapper } from '@vue/test-utils'

const CopyButton = {
  template: `<z-button color="secondary" iconStyle="p-1" icon="copy" iconSize="small" iconColor="vanilla-100"></z-button>`,
  components: {
    ZButton
  }
}

describe('Textarea Component', () => {
  let mountFn: (options?: Record<string, unknown>) => Wrapper<Vue>
  let localVue: typeof Vue

  beforeEach(() => {
    mountFn = (options = {}) => {
      return mount(ZTextarea, {
        localVue,
        ...options
      })
    }
  })

  it('renders the default textarea component', () => {
    const wrapper = mountFn()

    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.findAll('textarea').length).toEqual(1)
  })

  it('renders a disabled textarea', () => {
    const wrapper = mountFn({
      propsData: {
        disabled: true
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.find('textarea').attributes('disabled')).toMatch('disabled')
  })
  it('renders a textarea with right slot', () => {
    const wrapper = mountFn({
      slots: {
        right: CopyButton
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
