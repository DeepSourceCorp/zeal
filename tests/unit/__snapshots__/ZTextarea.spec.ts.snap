// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textarea Component renders a disabled textarea 1`] = `<div class="h-full overflow-scroll p-1 border border-ink-200 flex items-start focus-within:shadow-white rounded-sm cursor-not-allowed resize-none"><textarea placeholder="Once upon a time..." disabled="disabled" class="w-full h-full p-1 bg-transparent resize-none outline-none text-xs text-slate cursor-not-allowed"></textarea> </div>`;

exports[`Textarea Component renders a textarea with right slot 1`] = `
<div class="h-full overflow-scroll p-1 border border-ink-200 flex items-start focus-within:shadow-white rounded-sm resize-none"><textarea placeholder="Once upon a time..." class="w-full h-full p-1 bg-transparent resize-none outline-none text-xs text-vanilla-400"></textarea> <button href="#" type="button" class="z-btn inline-flex items-center justify-center font-medium transition-colors duration-300 ease-in-out rounded-sm focus:outline-none whitespace-nowrap p-0 h-10 w-10 bg-ink-300 text-vanilla-100 hover:bg-ink-200"><svg viewBox="0 0 24 24" stroke-linecap="round" stroke-linejoin="round" fill="none" class="stroke-1.5 z-icon w-4 h-4 z-icon--copy text-text-vanilla-100">
      <rect x="9" y="9" width="13" height="13" rx="2" ry="2"></rect>
      <path d="M5 15H4a2 2 0 0 1-2-2V4a2 2 0 0 1 2-2h9a2 2 0 0 1 2 2v1"></path>
    </svg> </button></div>
`;

exports[`Textarea Component renders the default textarea component 1`] = `<div class="h-full overflow-scroll p-1 border border-ink-200 flex items-start focus-within:shadow-white rounded-sm resize-none"><textarea placeholder="Once upon a time..." class="w-full h-full p-1 bg-transparent resize-none outline-none text-xs text-vanilla-400"></textarea> </div>`;
